### DISCUSSION TITLE
What is cockpit using to measure interface bandwidth?

### DISCUSSION THREAD

**Discussion by JanoutV** [cockpit-project/cockpit#16907] [Q&A] [answered]
Hi,
I am new to cockpit and I really like the fact that cockpit is able to show network interface bandwidth which updates every second. I am curious, what tool does Cockpit use to achieve this?
![image](https://user-images.githubusercontent.com/44194415/152139780-fd9d4f73-8d21-4ed6-943e-27570f447fc4.png)

**Comment by martinpitt** [comment, marked as answer]
If you are asking in the context of "I want something similar on my own custom cockpit page": This is done in the [networking page](https://github.com/cockpit-project/cockpit/blob/main/pkg/networkmanager/helpers.js) with a [metrics channel](https://cockpit-project.org/guide/latest/cockpit-metrics.html) (see [protocol.md](https://github.com/cockpit-project/cockpit/blob/main/doc/protocol.md) for some details).

That metric is implemented in [cockpit-bridge](https://github.com/cockpit-project/cockpit/blob/main/src/bridge/cockpitnetworksamples.c) by polling `/proc/net/dev`.
